app/core/views/ticket.py

```diff
@@ -0,0 +1,219 @@
+import markdown
+
+from django.http import Http404
+from django.urls import reverse
+from django.views import generic
+
+from django_celery_results.models import TaskResult
+
+from access.mixin import OrganizationPermission
+
+from core.forms.ticket import DetailForm, TicketForm
+from core.models.ticket.ticket import Ticket
+from core.views.common import AddView, ChangeView, DeleteView, IndexView
+
+from settings.models.user_settings import UserSettings
+
+
+
+class Add(AddView):
+
+    form_class = TicketForm
+
+    model = Ticket
+
+
+    def get_dynamic_permissions(self):
+
+        return [
+            str('core.add_ticket_' + self.kwargs['ticket_type']),
+        ]
+
+
+    def get_initial(self):
+
+        initial = super().get_initial()
+
+        initial.update({
+            'type_ticket': self.kwargs['ticket_type'],
+        })
+
+        return initial
+
+
+    def form_valid(self, form):
+        form.instance.is_global = False
+        return super().form_valid(form)
+
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['request'] = self.request
+        return kwargs
+
+
+    def get_success_url(self, **kwargs):
+
+        if self.kwargs['ticket_type'] == 'request':
+
+            return reverse('Assistance:_ticket_request_view', args=(self.kwargs['ticket_type'],self.object.id,))
+        
+        else:
+
+            return reverse('ITIM:_ticket_' + str(self.kwargs['ticket_type']).lower() + '_view', args=(self.kwargs['ticket_type'],self.object.id,))
+
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['content_title'] = 'New Ticket'
+
+        return context
+
+
+
+class Change(ChangeView):
+
+    form_class = TicketForm
+
+    model = Ticket
+
+
+    def get_dynamic_permissions(self):
+
+        return [
+            str('core.change_ticket_' + self.kwargs['ticket_type']),
+        ]
+
+
+    def get_context_data(self, **kwargs):
+
+        context = super().get_context_data(**kwargs)
+
+        context['content_title'] = str(self.object)
+
+        return context
+
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['request'] = self.request
+        return kwargs
+
+
+    def get_initial(self):
```

Comment:
add test to all views to ensure that if this function is an override, it calls the parent. mock it.